riscv/src/compiler.rs

```diff
@@ -1428,6 +1433,40 @@ fn process_instruction(instr: &str, args: &[Argument]) -> Vec<String> {
             ]
         }
 
+        insn if insn.starts_with("lr.w") => {
+            // Very similar to "lw":
+            let (rd, rs, off) = rro(args);
+            assert_eq!(off, 0);
+            // TODO misaligned access should raise misaligned address exceptions
+            only_if_no_write_to_zero_vec(
+                vec![
+                    format!("addr <=X= {rs};"),
+                    format!("{rd} <== mload();"),
+                    // For consistency sake, I tried to use format!() here, but
+                    // the lint doesn't like it
+                    "lr_sc_reservation <=X= 1;".into(),
```

Comment:
Strictly, we should have one reservation bit per memory address (if I read the standard correctly). If we don't have that, maybe we could scrap the reservation altogether, or would some code break in that case?

Reply:
If we keep this, we should do it even if `rd` is zero.

Reply:
> Strictly, we should have one reservation bit per memory address (if I read the standard correctly). If we don't have that, maybe we could scrap the reservation altogether, or would some code break in that case?

The reservation can be for any memory size. It seems that a typical implementation is <= 1Kb, but the ISA specs says to assume it to be of any size, so ours is for the entire memory.

> If we keep this, we should do it even if `rd` is zero.

Indeed, makes sense.